lib/prefab/config_client.rb

```diff
@@ -253,10 +248,10 @@ def start_checkpointing_thread
     end
 
     def finish_init!(source, project_id)
-      return unless @initialization_lock.write_locked?
+      return if initialized?
 
       LOG.debug "Unlocked Config via #{source}"
-      @initialization_lock.release_write_lock
+      @initialization_lock.count_down
```

Comment:
Multiple threads can pass the initialized check, log and call count_down. I checked and the latch doesn't go below zero or care if its called multiple times though so may not be a big deal.
I recently put the equivalent code in the python client in a mutex to make it easier to reason about https://github.com/prefab-cloud/prefab-cloud-python/blob/main/prefab_cloud_python/config_client.py#L247